### PR TITLE
Paywall: limit to top level on posts, add icon and keywords

### PR DESCRIPTION
**Pull request by simison**
Based on findings in https://github.com/Automattic/jetpack/pull/32119#pullrequestreview-1552370730 

- Limit to top-level only, no inner blocks to avoid issues with visual rendering. We might relax this later but needs more attention before doing so to avoid issues like these:

    

- Use icon from "next page" block — we should invent our own but this is a pretty nice temporary one!

    image

- Add first keywords

## Proposed changes:

*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

- Add `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );` to your setup
- Create or Edit a post and add the paywall block
- Check icon
- Test adding as inner block (not allowed), and as top level block (allows)
- Test adding in post editor, page editor

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/paywall/index.js [hunk 1]
```diff
@@ -19,19 +19,27 @@ export const settings = {
...
+		_x( 'subscribe', 'block search term', 'jetpack' ),
+		_x( 'subscription', 'block search term', 'jetpack' ),
+		_x( 'subscriptions', 'block search term', 'jetpack' ),
+	],
 	supports: {
 		customClassName: false,
 		html: false,
 		multiple: false,
 	},
+	parent: [ 'core/post-content' ],
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/paywall/index.js:38]: Took from https://github.com/WordPress/gutenberg/blob/4b4914b7e4ccdeb6393e95eee4666c5e62a92af3/packages/block-library/src/nextpage/block.json#L9

**Comment by lezama** [comment]
> Use icon from "next page" block — we should invent our own but this is a pretty nice temporary one!

nice!

File: projects/plugins/jetpack/extensions/blocks/paywall/index.js (hunk 1 above)
Comment (lezama) [reply projects/plugins/jetpack/extensions/blocks/paywall/index.js:38]: perfect

**Review by lezama — APPROVED** [review]

File: projects/plugins/jetpack/extensions/blocks/paywall/index.js (hunk 1 above)
Comment (simison) [reply projects/plugins/jetpack/extensions/blocks/paywall/index.js:38]: Note that this just limits to root of the document, but we still need to restrict to just post-editor and exclude custom post types.

This hook seems good next step: https://developer.wordpress.org/reference/hooks/allowed_block_types_all/

**Pull request merged**